packages/vscode-dependencies-validation/test/autofix/eventUtilspec.ts

```diff
@@ -0,0 +1,69 @@
+import type { Uri } from "vscode";
+import * as proxyquire from "proxyquire";
+import { VscodeFileEventConfig } from "../../src/vscodeTypes";
+import { handlePackageJsonEvent } from "../../src/autofix/eventUtil";
+import { configurationProxy, utilProxy } from "../moduleProxies";
+import { SinonMock, createSandbox, SinonSandbox } from "sinon";
+
+describe("eventUtil unit tests", () => {
+  let sandbox: SinonSandbox;
+  const vscodeConfig = <VscodeFileEventConfig>{};
+  let configurationProxySinonMock: SinonMock;
+  let utilProxySinonMock: SinonMock;
+  let handlePackageJsonEventProxy: typeof handlePackageJsonEvent;
+
+  beforeEach(() => {
+    sandbox = createSandbox();
+
+    const eventUtilProxyModule = proxyquire("../../src/autofix/eventUtil", {
+      "../util": utilProxy,
+      "./configuration": configurationProxy,
+    });
+
+    utilProxySinonMock = sandbox.mock(utilProxy);
+    configurationProxySinonMock = sandbox.mock(configurationProxy);
+    handlePackageJsonEventProxy = eventUtilProxyModule.handlePackageJsonEvent;
+  });
+
+  afterEach(() => {
+    utilProxySinonMock.verify();
+    configurationProxySinonMock.verify();
+  });
+
+  context("handlePackageJsonEvent()", () => {
+    it("autofix is disabled", async () => {
+      const uri = <Uri>{ fsPath: "root/folder/project/package.json" };
+      configurationProxySinonMock.expects("isAutoFixEnabled").returns(false);
+      utilProxySinonMock.expects("findAndFixDepsIssues").never();
+
+      await handlePackageJsonEventProxy(uri, vscodeConfig);
+    });
+
+    it("path is in node_modules", async () => {
```

Comment:
I don't think this is a **maintainable** test.

- It seems to test an **flow** instead of being "unity".
- It utilizes very specific knowlage (whitebox) of which methods would be called **trasitivily** due to the invoking  `handlePackageJsonEventProxy`
- It "translates" this **whitebox** information on the proxied methods expected return values 
  to test assertions that "prove" correct behaivor.
- and it refers to these methods via **string** names, so future refactoring would break the test.

Maybe the events flows are not easily testable without more E2E / Integration testing capabilities.
We need to investigate this, but I suspect this style of test will cause high TCO over time.

Reply:
Let's talk about it.
In my opinion this is the only "unit test" that we can implement here